core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -79,6 +82,13 @@ private[spark] abstract class Task[T](
       metrics)
     TaskContext.setTaskContext(context)
     taskThread = Thread.currentThread()
+
+    val callerContext =
+      s"Spark_AppId_${appId.getOrElse("")}_AppAttemptId_${appAttemptId.getOrElse("None")}" +
+        s"_JobId_${jobId.getOrElse("0")}_StageID_${stageId}_stageAttemptId_${stageAttemptId}" +
+        s"_taskID_${taskAttemptId}_attemptNumber_${attemptNumber}"
```

Comment:
one concern I have about this is the length of the string.  This string is going to be sent on every hadoop RPC call and then I believe get into the hdfs audit log.  The audit log does have a config to truncate it (default 128) but it still gets sent over rpc.  So I would like to keep this string as small as possible while still being useful.  The one above seems very long to me since just the static string is like 90 characters.  So I would like to see some of those static strings removed or abbreviated.